app/controllers/home_controller.rb

```diff
@@ -21,4 +21,38 @@ def post_comments
 			redirect_to root_url
     end
   end
+
+  def like_post_and_comment
+    @like = Like.find_by_likeable_id_and_likeable_type_and_user_id(params[:likeable_id], params[:likeable_type], current_user.id)
+    if !@like.present?
+      @like = current_user.likes.new(likeable_type: params[:likeable_type], likeable_id: params[:likeable_id])
+      respond_to do |format|
```

Comment:
You controller action should never exceed more than **10-12** lines of code
`respond_to` is used once per action so no need to use it twice here 

1. First find if you can find the `like` with given params 
    - if not then create a new one
    - if you find the like then destroy it
2. return same response at the end and NOTE: it should only respond to JS for now as far as I know we're not going to use it to respond against HTML request

If still unclear we can have a call and discuss it over there